Source/cryptalgo/SecureSocketPort.h

```diff
@@ -72,19 +72,26 @@ namespace Crypto {
             };
 
         public:
+            // Type of underlying socket
+            enum class sockettype_t : uint8_t {
+                   CLIENT_SOCKET
+                ,  SERVER_SOCKET
+            };
+
             Handler(Handler&&) = delete;
             Handler(const Handler&) = delete;
             Handler& operator=(const Handler&) = delete;
 
             template <typename... Args>
-            Handler(SecureSocketPort& parent, Args&&... args)
+            Handler(SecureSocketPort& parent, sockettype_t type, Args&&... args)
                 : Core::SocketPort(args...)
                 , _parent(parent)
                 , _context(nullptr)
                 , _ssl(nullptr)
                 , _callback(nullptr)
-                , _handShaking(CONNECTING) {
-            }
+                , _handShaking(type == sockettype_t::CLIENT_SOCKET ? CONNECTING : ACCEPTING)
```

Comment:
_handShaking is properly initialized during the Initialize() call t CONNECTING or ACCEPTING, based on the state of the socket if it is being called. Suggest to set it to ERROR here to assert in the INITIALIZE if, for example, it does not have this vale if we "Initialize()". Making sure the Initialize is called only once..

Reply:
It was not the intended goal as mentioned here https://github.com/rdkcentral/Thunder/pull/1796#discussion_r1858014895

Reply:
Not the state of the socket as in open or closed. The intent is to differentiate between and identify the owner of the socket: a server using the accept method to construct it, or, a client that creates it to connect to a server, Handler::ACCEPTING, Handler::CONNECTING respectively.